### DISCUSSION TITLE
HAA 11 => 12.7.3 migration

### DISCUSSION THREAD

**Discussion by Anton1oK** [RavenSystem/esp-homekit-devices#2108] [General]
Hi! I'm trying to update the following code to HAA v.12 and want to ask for help.
Here is a code I was using for a PC power button.
`{"c":{"l":13,"b":[{"g":0,"t":5}]},"a":[{"0":{"r":[{"g":12}]},"1":{"r":[{"g":12,"v":1}]},"b":[{"g":0}]}]}`

Here is what I got trying to update the code: 
`{"c":{"io":[[[12],2],[[0],6,1],[[13],3]],"l":13,"b":[{"g":0,"t":5}]},"a":[{"0":{"r":[{"g":12}]},"1":{"r":[{"g":12,"v":1}]},"b":[{"g":0}]}]}`

Please check if I'm missing something
I really appreciate any help you can provide!

**Comment by pasparakis** [comment]
{"c":{"l":13,"b":[[0,5]],"io":[[[13],3],[[0],6],[[12],2]]},"a":[{"0":{"r":[[12]]},"1":{"r":[[12,1]]},"b":[[0]]}]}

I'm not sure...try it..

**Reply by Anton1oK** [reply]
Thank you! That works great!
